cloud/libvirt/actuators/machine/utils/volume.go

```diff
@@ -196,46 +194,55 @@ func CreateVolume(volumeName, poolName, baseVolumeID, source, volumeFormat strin
 	return nil
 }
 
-// DeleteVolume removes the volume identified by `key` from libvirt
-func DeleteVolume(name string, client *Client) error {
-	volumePath := fmt.Sprintf(baseVolumePath+"%s", name)
-	volume, err := client.libvirt.LookupStorageVolByPath(volumePath)
+func getVolumeFromPool(volumeName, poolName string, virConn *libvirt.Connect) (*libvirt.StorageVol, error) {
+	pool, err := virConn.LookupStoragePoolByName(poolName)
 	if err != nil {
-		return fmt.Errorf("Can't retrieve volume %s", volumePath)
+		return nil, fmt.Errorf("can't find storage pool %q: %v", poolName, err)
 	}
-	defer volume.Free()
+	defer pool.Free()
 
 	// Refresh the pool of the volume so that libvirt knows it is
 	// not longer in use.
-	volPool, err := volume.LookupPoolByVolume()
+	waitForSuccess("error refreshing pool for volume", func() error {
+		return pool.Refresh(0)
+	})
+
+	// Check whether the storage volume exists. Its name needs to be
+	// unique.
+	volume, err := pool.LookupStorageVolByName(volumeName)
 	if err != nil {
-		return fmt.Errorf("Error retrieving pool for volume: %s", err)
+		return nil, fmt.Errorf("can't retrieve volume %q: %v", volumeName, err)
 	}
-	defer volPool.Free()
+	//defer volume.Free()
```

Comment:
Do we need to remove this and instead free the returned volumes in each of the calling sites?